Plugins/Player/Player.cpp

```diff
@@ -1064,4 +1071,74 @@ ArgumentStack Player::GetQuestCompleted(ArgumentStack&& args)
     return stack;
 }
 
+void Player::LoadCharacterFinishHook(NWNXLib::Services::Hooks::CallType cType,
+                                       NWNXLib::API::CServerExoAppInternal *,
+                                       NWNXLib::API::CNWSPlayer* pPlayer,
+                                       int32_t,
+                                       int32_t)
+{
+    if (cType == Services::Hooks::CallType::AFTER_ORIGINAL && !g_plugin->m_PersistentLocationWP.empty())
+    {
+        std::string sKey;
+        std::string sResRef = std::string(pPlayer->m_resFileName.GetResRef(), pPlayer->m_resFileName.GetLength());
+        if (Globals::AppManager()->m_pServerExoApp->GetServerInfo()->m_PersistantWorldOptions.bServerVaultByPlayerName)
+        {
+            std::string sCommunityName = pPlayer->GetPlayerName().CStr();
+            sKey = sCommunityName + "!" + sResRef;
+        }
+        else
+        {
+            auto *pNetLayer = Globals::AppManager()->m_pServerExoApp->GetNetLayer();
+            auto *pPlayerInfo = pNetLayer->GetPlayerInfo(pPlayer->m_nPlayerID);
+            std::string sCDKey = pPlayerInfo->GetPublicCDKey(0).CStr();
+            sKey = sCDKey + "!" + sResRef;
+        }
+        auto wpOID = g_plugin->m_PersistentLocationWP[sKey].first;
+        if (!wpOID)
+            return;
+
+        auto bFirstConnectOnly = g_plugin->m_PersistentLocationWP[sKey].second;
+
+        // Delete the key if this is the first connect and we're only setting the location on first connect
+        if (bFirstConnectOnly && !Utils::GetModule()->GetPlayerTURDFromList(pPlayer))
+        {
+            g_plugin->m_PersistentLocationWP.erase(sKey);
+        }
+        // The TURD exists already meaning its not the first connect
+        else if (bFirstConnectOnly)
+            return;
+
+        // Fake some changes to their area/position as though they had a TURD
+        auto *pWP = Utils::AsNWSWaypoint(Utils::GetGameObject(wpOID));
+        if (pWP)
+        {
+            auto pCreature = Utils::AsNWSCreature(Utils::GetGameObject(pPlayer->m_oidNWSObject));
+            pCreature->m_oidDesiredArea = pWP->m_oidArea;
+            pCreature->m_vDesiredAreaLocation = pWP->m_vPosition;
+            pCreature->m_bDesiredAreaUpdateComplete = false;
+            pCreature->m_vOrientation = pWP->m_vOrientation;
+            pPlayer->m_bFromTURD = true;
+            // We don't need the WP any more if we're only setting the location on first connect
+            if (bFirstConnectOnly)
+            {
+                pWP->RemoveFromArea();
```

Comment:
Mind, this doesn't actually destroy the WP, it just lives somewhere in "limbo". I wonder if you could do a `delete pWP;` after `RemoveFromArea()` without the game exploding.